psamm/datasource/sbml.py

```diff
@@ -711,6 +711,46 @@ def _get_flux_bounds(self, r_id, model, flux_limits, equation):
             upper_str = text_type(flux_limits[r_id][1])
         return lower_str, upper_str
 
+    def _add_gene_associations(self, r_id, r_genes, gene_ids, r_tag):
+        """Adds all the different kinds of genes into a list."""
+        if r_id in r_genes:
+            genes = ET.SubElement(
+                r_tag, _tag('geneProductAssociation', FBC_V2))
+            if isinstance(r_genes[r_id], list):
+                e = Expression(
+                    And(*(Variable(i) for i in r_genes[r_id])))
+            else:
+                e = Expression(r_genes[r_id])
+            gene_stack = [(e._root, genes)]
+            while len(gene_stack) > 0:
+                current, parent = gene_stack.pop()
+                if isinstance(current, Or):
+                    gene_tag = ET.SubElement(parent, _tag('or', FBC_V2))
+                elif isinstance(current, And):
+                    gene_tag = ET.SubElement(parent, _tag('and', FBC_V2))
+                elif isinstance(current, Variable):
+                    gene_tag = ET.SubElement(parent, _tag(
+                        'geneProductRef', FBC_V2))
+                    if current.symbol not in gene_ids:
+                        id = 'g_' + self._create_unique_id(
+                            gene_ids, self._make_safe_id(current.symbol))
+                        gene_ids[id] = current.symbol
+                    gene_tag.set(_tag('geneProduct', FBC_V2),
+                                 gene_ids[id])
+                if isinstance(current, (Or, And)):
+                    for item in current:
+                        gene_stack.append((item, gene_tag))
+        return gene_ids
+
+    def _add_gene_list(self, parent_tag, gene_id_list):
+        """ List of all gene products as sbml readable elements."""
```

Comment:
Remove initial whitespace in docstring. I would change the name of `gene_id_list` since it is not a list (maybe `gene_id_dict`). Docstring should start with a verb (maybe `Create list of all gene product...`).